### ISSUE TITLE
work remaining to deploy to GovCloud 

### ISSUE THREAD

**Issue by EricSchles** [18F/analytics-reporter#172]
1) Fix issues with publishing to s3 bucket
2) Debug and fix `analytics` script
3) setup cron tasks 

This touches https://github.com/18F/analytics.usa.gov as well

**Comment by alain-hoang** [comment]
When he `analytics` script is run it will publish the data with a `.json` file extension in a compressed gzip format.  Viewing the contents of downloaded data might require uncompressing the data in order to look at the internal contents.  The following code snippet illustrates how the data will be sent to the AWS bucket :

    var publish = function(name, data, extension, options, callback) {
      if (options.debug) console.log("[" + name + "] Publishing to " + config.aws.bucket + "...");
    
      var mime = {".json": "application/json", ".csv": "text/csv"};
    
      zlib.gzip(data, function(err, compressed) {

**Comment by alain-hoang** [comment]
Reminder to delete 18f pages yaml  (_config/18f.yaml)

**Comment by alain-hoang** [comment]
`Report ids` for Analytics can be confirmed by looking in the analytics dashboard under the Admin section

**Comment by rogeruiz** [comment]
The acceptance criteria to close this issue should be:

- [ ] Document or update existing documentation regarding the bucket file directory structure
- [ ] Document or update existing documentation regarding the solution to step 2
- [ ] Document or update existing documentation regarding the updated cron task 
- [x] Update the paths to GovCloud
- [x] Update credentials for cron
- [x] Update `exec` commands to point to the relative file path
- [x] Ensure there are manifests

**Comment by rogeruiz** [comment]
PR in progress => 18F/analytics.usa.gov#404

**Comment by EricSchles** [comment]
_development.yml from analytics.usa.gov seems to be working only on index.html (the main page for analytics.usa.gov) which is confusing.  Given that the rest of the configuration comes from _config.yml.

**Comment by alain-hoang** [comment]
The following items are being documented as possible follow up items that would require more engineering effort in order to setup:

- [ ]  Update paths in the cron.js script to use proper relative pathname resolution.    The paths are are set to work in a static manner at present
- [ ]  Update so executed scripts from cron.js are logging to properly location.   Logs are not actively logging at this time.
- [ ]  Setup, bind and verify a user provided environment service so setting such as environment variables can be set in the cloud.gov environment

**Comment by rogeruiz** [comment]
Docs living in #173

**Comment by konklone** [comment]
From my vantage point, this is what still needs to be done:

- [ ] cron.js should use relative pathnames for disk locations, perhaps based on an env var.
- [ ] Log executed scripts from cron.js to the proper location.
- [ ] Move the Google Analytics key data into an ENV var -- do not require the key file to be present on disk during the `cf push` process.
- [ ] Remove the `deploy-local` dir and `deploy` dir, then rename `deploy-GovCloud` to `deploy`. Update other paths to match.

**Comment by jmhooper** [comment]
@konklone:

What is the work that needs to be done for "Log executed scripts from cron.js to the proper location."?

**Comment by jmhooper** [comment]
Also, thoughts on moving the checkboxes related to managing keys into a separate issue then closing this once #181 is merged? To me those seem related to a separate piece of work that warrants its own discussion.

**Comment by konklone** [comment]
> What is the work that needs to be done for "Log executed scripts from cron.js to the proper location."?

I think it's that the cron.js scripts send logs of their shell actions to a file, instead of to stdout (where you can watch the action with `cf logs`).

> Also, thoughts on moving the checkboxes related to managing keys into a separate issue then closing this once #181 is merged? To me those seem related to a separate piece of work that warrants its own discussion.

Totally, :+1:.

**Comment by jmhooper** [comment]
I'll open a new issue for environment management and close this once the outputs for `cron.js` are logging to `stdout`.

**Comment by jmhooper** [comment]
I've opened #183 to handle moving keys into a cloud.gov user provided service which should cover the last 4 points.
